src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1075,19 +1085,26 @@ def dbapi(
             snowflake_table_name = random_name_for_temp_object(TempObjectType.TABLE)
             create_table_sql = (
                 "CREATE "
-                + f"{get_temp_type_for_object(self._session._use_scoped_temp_objects, True) if snowflake_table_type.lower() in TEMPORARY_STRING_SET else snowflake_table_type} "
-                + "TABLE "
-                + f"{snowflake_table_name} "
-                + f"""({" , ".join([f'"{field.name}" {convert_sp_to_sf_type(field.datatype)} {"NOT NULL" if not field.nullable else ""}' for field in struct_schema.fields])})"""
+                f"{get_temp_type_for_object(self._session._use_scoped_temp_objects, True) if snowflake_table_type.lower() in TEMPORARY_STRING_SET else snowflake_table_type} "
+                "TABLE "
+                f"{snowflake_table_name} "
+                f"""({" , ".join([f'"{field.name}" {convert_sp_to_sf_type(field.datatype)} {"NOT NULL" if not field.nullable else ""}' for field in struct_schema.fields])})"""
+                f"""{DATA_SOURCE_SQL_COMMENT}"""
```

Comment:
DATA_SOURCE_SQL_COMMENT is this what make telemetry be able to see that this SQL come out of data source api?

Reply:
the major way to track in this PR is using the statement_parameters.
this is a back-up approach marking the sql query